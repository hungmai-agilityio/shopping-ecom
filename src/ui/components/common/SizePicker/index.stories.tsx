import { useState } from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { colors, sizeInch, sizes } from '@/mock';

import { SizePicker } from '@/ui/components';

const meta: Meta<typeof SizePicker> = {
  component: SizePicker,
  parameters: {
    layout: 'centered'
  },
  tags: ['autodocs'],
  argTypes: {
    size: {
      description: 'size item'
    },
    selected: {
      description: 'Size that is currently selected'
    },
    onClick: {
      description: 'Callback when color changes',
      action: 'colorChanged'
    }
  }
};

export default meta;

type Story = StoryObj<typeof SizePicker>;

export const primary: Story = {
  render: (args) => {
    const [selectedSize, setSelectedSize] = useState<string>('');

    const handleSelect = (size: string) => {
      setSelectedSize(size);
    };
    return (
      <SizePicker
        size={args.size}
        selected={selectedSize}
        onClick={handleSelect.bind(null, sizes[0])}
      />
    );
  },
  args: {
    size: sizes[0]
  }
};

export const listSize: Story = {
  render: () => {
    const [selectedSize, setSelectedSize] = useState<string>(sizeInch[0]);

    const handleSelect = (size: string) => {
      setSelectedSize(size);
    };
    return (
      <div className="flex gap-4">
        {sizeInch.map((size, index) => (
          <SizePicker
            key={index}
            size={size}
            selected={selectedSize}
            onClick={handleSelect.bind(null, size)}
          />
        ))}
      </div>
    );
  }
};
