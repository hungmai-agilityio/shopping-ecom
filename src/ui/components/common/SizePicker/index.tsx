import { clsx } from 'clsx';

// Constants
import { popping } from '@/constants';

interface SizeProps {
  size: string;
  selected: string;
  onClick: () => void;
}

const SizePicker = ({ size, selected, onClick }: SizeProps) => {
  return (
    <button
      onClick={onClick}
      className={clsx(
        'min-w-8 h-8 px-1 max-w-btn-size truncate border border-dark rounded text-base hover:bg-primary hover:text-white hover:border-primary',
        popping.className,
        {
          'bg-primary text-white border-primary': selected === size
        }
      )}
    >
      {size}
    </button>
  );
};

export default SizePicker;
