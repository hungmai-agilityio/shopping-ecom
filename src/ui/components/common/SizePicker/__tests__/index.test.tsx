import SizePicker from '@/ui/components/common/SizePicker';
import { render, screen } from '@testing-library/react';

describe('SizePicker component', () => {
  const defaultProps = {
    size: 'S',
    selected: '',
    onClick: jest.fn()
  };

  test('Should render component with default props', () => {
    render(<SizePicker {...defaultProps} />);

    const size = screen.getByText('S');

    expect(size).toBeInTheDocument();
    expect(size).toHaveClass(
      'min-w-8 min-h-8 px-1 max-w-btn-size truncate border border-dark rounded text-base'
    );
  });

  test('Should render component with selected item', () => {
    render(<SizePicker {...defaultProps} selected="S" />);

    const size = screen.getByText('S');

    expect(size).toBeInTheDocument();
    expect(size).toHaveClass('bg-primary text-white border-primary');
  });

  test('Should be render match to snapshot', () => {
    const { container } = render(<SizePicker {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
