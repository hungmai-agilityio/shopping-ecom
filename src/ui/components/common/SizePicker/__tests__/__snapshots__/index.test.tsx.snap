// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizePicker component Should be render match to snapshot 1`] = `
<div>
  <button
    class="min-w-8 min-h-8 px-1 max-w-btn-size truncate border border-dark rounded text-base hover:bg-primary hover:text-white hover:border-primary className"
  >
    S
  </button>
</div>
`;
